lib/resque/worker.rb

```diff
@@ -226,33 +227,34 @@ def work(interval = 5.0, &block)
           working_on job
 
           procline "Processing #{job.queue} since #{Time.now.to_i} [#{job.payload_class_name}]"
-          if @child = fork(job)
-            srand # Reseeding
-            procline "Forked #{@child} at #{Time.now.to_i}"
+          if fork_per_job?
+            run_hook :before_fork, job
             begin
-              Process.waitpid(@child)
-            rescue SystemCallError
-              nil
+              @child = fork do
+                unregister_signal_handlers if term_child
+                perform(job, &block)
+                exit! unless run_at_exit_hooks
```

Comment:
This used to say `run_at_exit_hooks ? exit : exit!`. Where did the `exit` (without the `!` go?). Is that unnecessary now because the child exits automatically when the `fork` block ends?